tests/consent_mailer/conftest.py

```diff
@@ -45,19 +45,9 @@ def mocked_consent_backend_api_connector(monkeypatch: pytest.MonkeyPatch) -> Non
         )
     ]
 
-    class FakeConnector:
-        def assign_identity(
-            self,
-            had_primary_source: MergedPrimarySourceIdentifier,
-            identifier_in_primary_source: str,
-        ) -> Identity:
-            return Identity(
-                identifierInPrimarySource=identifier_in_primary_source,
-                identifier="efVXbcoxxMAiA5IqT1d11H",
-                stableTargetId="efVXbcoxxMAiA5IqT1d11S",
-                hadPrimarySource=had_primary_source,
-            )
+    call_counter = {"count": 0}
 
+    class FakeConnector:
         def fetch_all_merged_items(
```

Comment:
Instead of calling `call_counter["count"] += 1` in every branch u could just call it once in the first line of the function.

```suggestion
    ...
    ) -> Generator[AnyMergedModel, None, None]:
        call_counter["count"] += 1
        if (
            entity_type
            and "MergedPerson" in entity_type
            and reference_field == "hadPrimarySource"
        ):
        ...
    ...
```